packages/service-library/src/servicelib/rabbitmq.py

```diff
@@ -72,13 +90,29 @@ def __post_init__(self):
         # channels are not thread safe, what about python?
         self._channel_pool = aio_pika.pool.Pool(self._get_channel, max_size=10)
 
+    async def rpc_initialize(self) -> None:
+        # TODO: to SAN: not sure that we always want to setup RPC connection
+        self._rpc_connection = await aio_pika.connect_robust(
+            self.settings.dsn, client_properties={"connection_name": f"rpc.{uuid4()}"}
+        )
+        self._rpc_channel = await self._rpc_connection.channel()
+        self._rpc = await RPC.create(self._rpc_channel)
+
     async def close(self) -> None:
         with log_context(log, logging.INFO, msg="Closing connection to RabbitMQ"):
             assert self._channel_pool  # nosec
             await self._channel_pool.close()
             assert self._connection_pool  # nosec
             await self._connection_pool.close()
 
+            # rpc is not always initialized
+            if self._rpc is not None:
```

Comment:
Does the order matter? Can it be done in parallel? 
What if one close raises exception?

Reply:
The order matters and is correct. Closing them in opposite order of creation.
If an exception is raised, most likely there is an error int the close sequence. Library error.